### ISSUE TITLE
About the aul_loss

### ISSUE THREAD

**Issue by Bian-jh** [InterDigitalInc/CompressAI#305]
Can I add aul_loss to rate_distortion_loss as a total loss and then use one optimizer to optimize the entire network?

**Comment by YodaEmbedding** [comment]
- When minimizing `loss` (i.e. training the main network) , `EntropyBottleneck.quantiles` is unused, and therefore receives no gradients.
- Minimizing `aux_loss` trains only the `EntropyBottleneck.quantiles`.

The above two losses and their gradient updates are therefore independent. That is, performing one optimizer step for minimizing `loss` and one optimizer step for minimizing `aux_loss` is equivalent to performing one optimizer step of `loss + aux_loss`. (Assuming the LRs are equal.)

So... yes.

**Comment by Bian-jh** [comment]
> * When minimizing `loss` (i.e. training the main network) , `EntropyBottleneck.quantiles` is unused, and therefore receives no gradients.
> 
>     * Minimizing `aux_loss` trains only the `EntropyBottleneck.quantiles` (since `stop_gradient=True`).
> 
> 
> The above two losses and their gradient updates are therefore independent. That is, performing one optimizer step for minimizing `loss` and one optimizer step for minimizing `aux_loss` is equivalent to performing one optimizer step of `total_loss = loss + aux_loss`. (Assuming the LRs are equal.)
> 
> So... yes.

Thank you for your immediate reply!！
